docs/source/notes/extending.rst

```diff
@@ -1,10 +1,238 @@
 Extending PyTorch
 =================
 
-In this note we'll cover ways of extending :mod:`torch.nn`,
+In this note we'll cover ways of extending :mod:`torch`, :mod:`torch.nn`,
 :mod:`torch.autograd`, and writing custom C extensions utilizing our C
 libraries.
 
+Extending :mod:`torch`
+----------------------
+
```

Comment:
I think a little more context setting here would be helpful for readers. Something like, "You can create Tensor-like objects in PyTorch by defining your own class with methods defined appropriately. But what if you want to override top-level torch functions like `torch.add`?"